Add support for config file

mqnguyen5 opened the pull request:
Hello,

I have make some adjustments to the existing code to support the new feature:

- Installed [jsonfile](https://www.npmjs.com/package/jsonfile) package to read & parse JSON file
- Added `-c` option to handle config files in `./bin/index.js`
- Added support for Markdown files when the input is a folder in `./bin/utils/processFolder.js`
- Added new module, `./bin/utils/processJSONFile.js` to read and process inputs from JSON file
- Added sample `sample.json` file for testing

minhhang107 commented on bin/utils/processFolder.js:
```diff
@@ -19,9 +19,20 @@ const processFolder = (input, output, stylesheet) => {
         processFile(file, output, stylesheet);
       });
 
+    const mdFiles = files.filter((file) => path.extname(file) === ".md");
```
We could combine with the 2 filters for .md and .txt into 1 filter only
eg: `const supportedFiles = files.filter((...)=> ... === ".txt" || ... === ".md")`

minhhang107 commented on bin/utils/processJSONFile.js:
```diff
@@ -0,0 +1,73 @@
+      return process.exit(1);
+    }
+
+    fs.lstat(data.input, (err, stats) => {
```
This chunk (`ls.lstat(){}`) is duplicating the `index.js`. We could separate it into a new module (eg: `processInput`) to reuse.

minhhang107 commented on bin/utils/processJSONFile.js:
```diff
@@ -0,0 +1,73 @@
+        );
+        return process.exit(1);
+      }
+
```
the `output` and `stylesheet` should be moved out of `ls.lstat()` so we could modularize this part

minhhang107 commented on sample.json:
```diff
@@ -0,0 +1,5 @@
+{
```
I don't give sample files for other features so let's keep it consistent by removing `sample.json` or adding it to `.gitignore`

minhhang107 reviewed (changes requested):
The new feature looks great but I have a few suggestions. Please see comments in each file for details.

mqnguyen5 commented:
I have added new changes based on the suggestions, specifically:

- Added new module `/bin/utils/processInput.js` to handle input processing logic
- Moved input processing logic from `/bin/index.js` and `/bin/utils/processJSONfile.js` to new module and cleaned up unused modules/variables
- Removed `sample.json`
- Combine file processing logic in `/bin/utils/processFolder.js` to handle both text and markdown files

If possible, please review the changes as soon as possible and let me know if there is anything else needs fixing.
